src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -2253,10 +2264,14 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                                     nullptr,
                                     m_storage, is_array_type, is_malloc_array_type,
                                     is_list, m_dims, n_dims, a_kind));
+            tuple_type->m_type[tuple_type_left->n_type + i] = tuple_type_right->m_type[i];
         }
         llvm::Type* concat_tuple_type = tuple_api->get_tuple_type(type_code, llvm_el_types);
         llvm::Value* concat_tuple = builder->CreateAlloca(concat_tuple_type, nullptr, "concat_tuple");
-        tuple_api->concat(left, right, tuple_type_left, tuple_type_right, concat_tuple);
+        tuple_api->concat(left, right, tuple_type_left, tuple_type_right, concat_tuple,
+                          tuple_type, *module, name2memidx);
+        delete[] tuple_type->m_type;
+        delete tuple_type;
```

Comment:
These `delete` can be removed, once you use `make_Tuple_t`.

Also in general, never use raw `new` and `delete` in modern C++. Always use some kind of a smart pointer or allocator.